modules/vmdatasource/test/test_remove.cpp

```diff
@@ -315,6 +324,138 @@ TEST_F(TestRemovingSchema, RemoveAll)
+        VMF_METADATA_END(newSchema);
+
+        newStream.reopen(vmf::MetadataStream::ReadWrite);
+        newStream.addSchema(newSchema);
+        newStream.save();
+        newStream.close();
+    }
+
+    {
+        vmf::MetadataStream newStream;
+        newStream.open(TEST_FILE, vmf::MetadataStream::ReadOnly);
+        bool result = newStream.load(TEST_SCHEMA_NAME);
+        ASSERT_TRUE( result );
+        std::shared_ptr<vmf::MetadataSchema> schema = newStream.getSchema(TEST_SCHEMA_NAME);
+        std::vector< std::shared_ptr<vmf::MetadataDesc> > descs = schema->getAll();
+        ASSERT_EQ(descs.size(), 1);
+        std::shared_ptr<vmf::MetadataDesc> desc = descs[0];
+        ASSERT_EQ(desc->getMetadataName(), TEST_PROPERTY_NAME2);
+        newStream.close();
+    }
+}
+
+
+TEST_F(TestRemovingSchema, RemoveOneAddSchemaBeforeSaving)
+{
+    {
+        vmf::MetadataStream newStream;
+        newStream.open(TEST_FILE, vmf::MetadataStream::ReadWrite);
+        newStream.load();
+        auto schemaForRemoving = newStream.getSchema(TEST_SCHEMA_NAME);
+        newStream.remove(schemaForRemoving);
+
+        std::shared_ptr<vmf::MetadataSchema> newSchema(new vmf::MetadataSchema(TEST_SCHEMA_NAME));
+        VMF_METADATA_BEGIN(TEST_PROPERTY_NAME2);
+            VMF_FIELD_INT(TEST_FIELD_NAME);
+        VMF_METADATA_END(newSchema);
+        newStream.addSchema(newSchema);
+
+        newStream.save();
+        newStream.close();
+    }
+
+    {
+        vmf::MetadataStream newStream;
+        newStream.open(TEST_FILE, vmf::MetadataStream::ReadOnly);
+        bool result = newStream.load(TEST_SCHEMA_NAME);
+        ASSERT_TRUE( result );
+        std::shared_ptr<vmf::MetadataSchema> schema = newStream.getSchema(TEST_SCHEMA_NAME);
+        std::vector< std::shared_ptr<vmf::MetadataDesc> > descs = schema->getAll();
+        ASSERT_EQ(descs.size(), 1);
+        std::shared_ptr<vmf::MetadataDesc> desc = descs[0];
+        ASSERT_EQ(desc->getMetadataName(), TEST_PROPERTY_NAME2);
+        newStream.close();
+    }
+}
+
+
+TEST_F(TestRemovingSchema, RemoveOneAddSchemaNoSaving)
+{
+    vmf::MetadataStream newStream;
+    newStream.open(TEST_FILE, vmf::MetadataStream::ReadWrite);
+    newStream.load();
+    auto schemaForRemoving = newStream.getSchema(TEST_SCHEMA_NAME);
+    newStream.remove(schemaForRemoving);
+
+    std::shared_ptr<vmf::MetadataSchema> newSchema(new vmf::MetadataSchema(TEST_SCHEMA_NAME));
+    VMF_METADATA_BEGIN(TEST_PROPERTY_NAME2);
+        VMF_FIELD_INT(TEST_FIELD_NAME);
+    VMF_METADATA_END(newSchema);
+    newStream.addSchema(newSchema);
+
+    std::shared_ptr<vmf::MetadataSchema> schema = newStream.getSchema(TEST_SCHEMA_NAME);
+    ASSERT_NE(schema, nullptr);
+    std::vector< std::shared_ptr<vmf::MetadataDesc> > descs = schema->getAll();
+    ASSERT_EQ(descs.size(), 1);
+    std::shared_ptr<vmf::MetadataDesc> desc = descs[0];
+    ASSERT_EQ(desc->getMetadataName(), TEST_PROPERTY_NAME2);
+    newStream.close();
```

Comment:
we also need to make sure that new schema is really stored into media file;
I suggest we call `save()`, `close()`, `clear()` then `re-open()`, `load()` and verify that data is correct - what's your opinion?